synapse/lib/stormlib/aha.py

```diff
@@ -1,22 +1,105 @@
-import synapse.common as s_common
 import synapse.lib.stormtypes as s_stormtypes
 
+@s_stormtypes.registry.registerLib
+class AhaLib(s_stormtypes.Lib):
+    '''
+    A Storm Library for interacting with AHA.
+    '''
+
+    _storm_locals = (
+        {'name': 'del', 'desc': 'Delete a service from AHA.',
+         'type': {'type': 'function', '_funcname': '_methSvcDel',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the service to delete.', },
+                      {'name': 'network', 'type': 'str', 'desc': 'An optional network to use when looking up the service name.',
+                       'default': None},
+                  ),
+                  'returns': {'type': 'null'}}},
+        {'name': 'get', 'desc': 'Get information about an AHA service.',
+         'type': {'type': 'function', '_funcname': '_methSvcGet',
+                  'args': (
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the AHA service to look up.', },
+                      {'name': 'filters', 'type': 'dict', 'desc': 'An optional dictionary of filters to use when resolving the AHA service.',
+                       'default': None}
+                  ),
+                  'returns': {'type': ('null', 'dict'), 'desc': 'The AHA service information dictionary, or $lib.null.', }}},
+        {'name': 'list', 'desc': 'List the services registered with AHA.',
+         'type': {'type': 'function', '_funcname': '_methSvcList', 'args': (),
+                  'returns': {'type': 'list', 'desc': 'A list of the AHA service dictionaries for all registered services.', }}},
+    )
+    _storm_lib_path = ('aha',)
+    def getObjLocals(self):
+        return {
+            'del': self._methSvcDel,
+            'get': self._methSvcGet,
+            'list': self._methSvcList,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methSvcList(self):
+        self.runt.reqAdmin()
+        proxy = await self.runt.snap.core.reqAhaProxy()
+        # DISCUSS: Do we want this to be a generator or list?
+        svcs = []
+        async for info in proxy.getAhaSvcs():
+            svcs.append(info)
+        return svcs
+
+    async def _methSvcDel(self, name, network=None):
```

Comment:
I'd prefer all exposed methods to either use absolute or `...` completed names rather than propagating the name/network schism issue.

Reply:
The delAhaSvc API requires a network value which matches that of the service's original registration.

The most straightforward way to address this would be to lookup the svcname provided; and then call the delAhaSvc API with the correct arguments.

That way we avoid having to change the Aha.delAhaSvc API.